xgcm/test/test_metric_ops_multi_axis.py

```diff
@@ -179,41 +173,35 @@ def test_missingaxis(self, axis, funcname, periodic, boundary):
             func = getattr(grid, funcname)
 
             if funcname == "cumint":
-                # cumint needs a boundary...
+                # cumint needs a boundary
                 kwargs = dict(boundary="fill")
             else:
                 kwargs = dict()
 
-            match_message = (
-                "Unable to find any combinations of metrics for array dims.*X.*Y.*Z.*"
-            )
+            match_message = "Did not find axis"
             with pytest.raises(KeyError, match=match_message):
-                func(ds, ["X", "Y", "Z"], **kwargs)
+                func(ds.tracer, ["X", "Y", "Z"], **kwargs)
 
-            match_message = (
-                "Unable to find any combinations of metrics for array dims.*X.*Y.*"
-            )
+            match_message = "Did not find axis"
             with pytest.raises(KeyError, match=match_message):
-                func(ds, ("X", "Y"), **kwargs)
+                func(ds.tracer, ("X", "Y"), **kwargs)
 
-    def test_missingdim(self, funcname, periodic, boundary):
+    def test_metric_axes_missing_from_array(self, funcname, periodic, boundary):
         ds, coords, metrics = datasets_grid_metric("C")
         grid = Grid(ds, coords=coords, metrics=metrics, periodic=periodic)
 
-        func = getattr(grid, funcname)
-
         if funcname == "cumint":
-            # cumint needs a boundary...
-            kwargs = dict(boundary=boundary)
+            # cumint needs a boundary
+            kwargs = dict(boundary="fill")
         else:
             kwargs = dict()
 
-        match_message = "Unable to find any combinations of metrics for array dims.*X.*"
-        with pytest.raises(KeyError, match=match_message):
+        func = getattr(grid, funcname)
+
+        match_message = "Did not find single matching dimension"
+        with pytest.raises(ValueError, match=match_message):
             func(ds.tracer.mean("xt"), "X", **kwargs)
 
-        match_message = (
-            "Unable to find any combinations of metrics for array dims.*X.*Y.*Z.*"
-        )
-        with pytest.raises(KeyError, match=match_message):
+        match_message = "Did not find single matching dimension"
+        with pytest.raises(ValueError, match=match_message):
```

Comment:
```suggestion
        with pytest.raises(ValueError, match="Did not find single matching dimension"):
```

Reply:
This post is relevant: http://matthewrocklin.com/blog/work/2019/06/23/avoid-indirection

Reply:
Thanks for this Deepak! I'll do my best to keep it in mind.